Yafc.UI/ImGui/ScrollArea.cs

```diff
@@ -74,20 +74,23 @@ public abstract class Scrollable(bool vertical, bool horizontal, bool collapsibl
             }
             else {
                 if (horizontal && maxScroll.X > 0f) {
-                    Rect fullScrollRect = new Rect(rect.X, rect.Bottom - ScrollbarSize, rect.Width, ScrollbarSize);
+                    Rect fullScrollRect = new Rect(rect.X, rect.Bottom, rect.Width, ScrollbarSize);
                     Rect scrollRect = new Rect(rect.X + scrollStart.X, fullScrollRect.Y, scrollSize.X, ScrollbarSize);
-                    BuildScrollBar(gui, 0, in fullScrollRect, in scrollRect);
+                    // The horizontal scrollbar is drawn '1 scrollbar' too high, so fix to match mouse click tests
+                    // helping the user to 'correctly' click the scrollbar.
+                    Rect drawRect = new Rect(scrollRect.X, scrollRect.Y - ScrollbarSize, scrollRect.Width, scrollRect.Height);
```

Comment:
Could you please elaborate on why introducing `drawRect` is necessary? I've read through the comment and the PR description, but still didn't get it.

Reply:
`drawRect` is a copy of `scrollRect` with the height adjusted (of the horizontal scrollbar), so it matches with the 'click testing area'. With this you click on the spot, so it is recognized as the scrollbar being clicked.

As mentioned, I have no idea exactly why this works and why it didn't work (I was just fed up with it, for years actually, and decided to try again to make the scrollbar work).

If someone has an idea/explanation why this is a good fix, or how it should be fixed better, let me know and we can improve the quality of this fix.

Reply:
The `drawRect` is gone now, so this comment thread is not relevant anymore.